contracts/contracts/utils/AerodromeAMOQuoter.sol

```diff
@@ -0,0 +1,316 @@
+            ) {
+                int24 currentTick;
+
+                // solhint-disable-next-line no-inline-assembly
+                assembly {
+                    currentTick := mload(add(reason, 0x24))
+                }
+                return
+                    RebalanceStatus({
+                        reason: RevertReasons.NotInExpectedTickRange,
+                        currentPoolWETHShare: 0,
+                        allowedWETHShareStart: 0,
+                        allowedWETHShareEnd: 0,
+                        currentTick: currentTick,
+                        balanceWETH: 0,
+                        amountWETH: 0,
+                        revertMessage: ""
+                    });
+            }
+
+            // Case 3: Not enough WETH for swap
+            if (
+                receivedSelector == IAMOStrategy.NotEnoughWethForSwap.selector
+            ) {
+                uint256 balanceWETH;
+                uint256 amountWETH;
+
+                // solhint-disable-next-line no-inline-assembly
+                assembly {
+                    balanceWETH := mload(add(reason, 0x24))
+                    amountWETH := mload(add(reason, 0x44))
+                }
+                return
+                    RebalanceStatus({
+                        reason: RevertReasons.NotEnoughWethForSwap,
+                        currentPoolWETHShare: 0,
+                        allowedWETHShareStart: 0,
+                        allowedWETHShareEnd: 0,
+                        currentTick: 0,
+                        balanceWETH: balanceWETH,
+                        amountWETH: amountWETH,
+                        revertMessage: ""
+                    });
+            }
+
+            // Case 4: Not enough WETH liquidity
+            if (
+                receivedSelector == IAMOStrategy.NotEnoughWethLiquidity.selector
+            ) {
+                return
+                    RebalanceStatus({
+                        reason: RevertReasons.NotEnoughWethLiquidity,
+                        currentPoolWETHShare: 0,
+                        allowedWETHShareStart: 0,
+                        allowedWETHShareEnd: 0,
+                        currentTick: 0,
+                        balanceWETH: 0,
+                        amountWETH: 0,
+                        revertMessage: ""
+                    });
+            }
+
+            // Case 5: Unexpected error
+            return
+                RebalanceStatus({
+                    reason: RevertReasons.UnexpectedError,
+                    currentPoolWETHShare: 0,
+                    allowedWETHShareStart: 0,
+                    allowedWETHShareEnd: 0,
+                    currentTick: 0,
+                    balanceWETH: 0,
+                    amountWETH: 0,
+                    revertMessage: abi.decode(reason, (string))
+                });
+        }
+    }
+}
+
+contract AerodromeAMOQuoter {
+    QuoterHelper public quoterHelper;
+
+    constructor() {
```

Comment:
constructor should probably also contain the `amoStrategy`  as a parameter and `clPool` (underlying aerodrome slipstream pool)